src/apps/service/apps/jobs/submissions.clj

```diff
@@ -179,65 +117,31 @@
 
 (defn- save-batch
   [user job-types app submission output-dir]
-  (let [batch-id (save-batch* user app submission output-dir)]
-    (save-batch-step batch-id (first job-types))
-    batch-id))
-
-(defn- map-slice
-  [m n]
-  (->> (map (fn [[k v]] (vector k (nth v n))) m)
-       (into {})
-       (remove-nil-values)))
-
-(defn- map-slices
-  [m]
-  (let [max-count (apply max (map (comp count val) m))]
-    (mapv (partial map-slice m) (range max-count))))
-
-
-(defn- substitute-param-values
-  [path-map config]
-  (->> (map (fn [[k v]] (vector k (get path-map v v))) config)
-       (into {})))
-
-(defn- format-submission-in-batch
-  [submission job-number path-map]
-  (let [job-suffix (str "analysis-" (inc job-number))]
-    (assoc (update-in submission [:config] (partial substitute-param-values path-map))
-      :group      (config/jex-batch-group-name)
-      :name       (str (:name submission) "-" job-suffix)
-      :output_dir (ft/path-join (:output_dir submission) job-suffix))))
-
-(defn- submit-job-in-batch
-  [apps-client user submission paths-exist job-number path-map]
-  (when (every? (partial get paths-exist) (map keyword (vals path-map)))
-    (let [job-info (.submitJob apps-client (format-submission-in-batch submission job-number path-map))]
-      (perms-client/register-private-analysis (:shortUsername user) (:id job-info))
-      job-info)))
-
-(defn- preprocess-batch-submission
-  [submission output-dir parent-id]
-  (assoc submission
-    :output_dir           output-dir
-    :parent_id            parent-id
-    :create_output_subdir false))
+  (transaction
+    (let [batch-id (save-batch* user app submission output-dir)]
+      (save-batch-step batch-id (first job-types))
+      (perms-client/register-private-analysis (:shortUsername user) batch-id)
+      batch-id)))
+
+(defn- pre-submit-batch-validation
+  [input-params-by-id input-paths-by-id path-list-stats]
+  (doseq [path-stat path-list-stats]
+    (validate-path-list-stats path-stat))
```

Comment:
I wonder if it might make sense to remove the path list file size validation and rely solely on the path list length validation. If we're going to modify the list length frequently then having to modify this setting in lockstep might be a bit of a pain. The one advantage to keeping this setting is that it allows us to constrain the amount of memory consumed by a single path list somewhat.

Reply:
For now I was going to set the file size limit to something large but not too large. A path list I've been using in testing has just over 30k paths and is just under 4MB, so I figure a limit of 64MB should be enough for anybody 😉  That lets us adjust the path limit without having to also adjust the size limit, but the size limit is there in case someone has very long paths.

I just realized that I could lookup the iRODS path length limit, and see how big of a file that would create with something like 100k paths. If that's reasonably small, then I agree that we can get rid of one limit and keep the other.

Reply:
After some discussion in chat with @ianmcorvidae, we decided we could get rid of the max HT Path file size limit as a config setting, and instead calculate a reasonable limit based on the max paths config setting. This helps prevent loading a massive, non-HT-Path-List file into memory by mistake (since users can set any file they wish as the HT Path List file type). In fact, it would also prevent loading a massive HT Path List file that goes way beyond the paths limit, and we wouldn't have to count the paths first before giving the client an error response.